### ISSUE TITLE
Compared to ridcp, how is this alleviate the texture distort.

### ISSUE THREAD

**Issue by loopsman** [Jiayi-Fu/IPC-Dehaze#7]
I find that this work presents higher detail and less texture distortion than ridcp. Could you explain it please?

**Comment by Jiayi-Fu** [comment]
1. ridcp indeed has feature loss due to VQ  during decoder feature alignment. Our method effectively reduces this loss by aligning the features of the encoder and decoder.
2. The token matching in ridcp is based on statistics, while our method uses a transformer for prediction. This prediction may yield better results. Additionally, we introduce a critic, which further enhances the overall effect.
You can refer to the data comparison in the paper regarding nn-matching (no iteration and critic), Ours (w/o Code-Critic), and Ours to see that our method gradually improves the effect based on ridcp.

**Comment by Jiayi-Fu** [comment]
I hope this answers your question🙂

**Comment by loopsman** [comment]
Thanks for your answer. I would like to further understand this excellent work,  allowing me to propose my question further: Did you utilize the codebook of RIDCP in this work？ And as I know, the codebook used by RIDCP is a basic version that has low utilization of the codebook, which means you only have a limited number of entries to match. So your work chose a better result than using Euclidean distance calculation within a limited range of options? If my understanding is incorrect, please let me know.

**Comment by Jiayi-Fu** [comment]
Due to the lack of discriminator weights in the publicly available pre-trained weights of RIDCP, we have retrained a basic vqgan (including codebook) in this work. Our vqgan network architecture and training method are roughly the same as those in RIDCP. Regarding your understanding, you are right. Although RIDCP has optimized the original method based on Euclidean distance matching through statistical methods, the matching capability still has limitations.

**Comment by loopsman** [comment]
Really appreciate your explanation and nice work.
